src/commands/interactive/untagged-tenants.ts

```diff
@@ -0,0 +1,311 @@
+
+    await meshAdapter.attachTenantCosts(allTenants, start, end);
+    return allTenants;
+  } else {
+    throw new MeshError("Something went horribly wrong.");
+  }
+}
+
+async function showUntagged(
+  options: CmdGlobalOptions,
+  data: MeshTenant[],
+  from: string | undefined = undefined,
+  to: string | undefined = undefined,
+  sortByCost: boolean,
+) {
+  console.clear();
+  const tags = getAllTags(data);
+  let running = true;
+
+  if (sortByCost) {
+    data = sortTenantDataByCost(options, data);
+  } else {
+    data = sortTenantDataByName(options, data);
+  }
+  while (running) {
+    const selectedTag = await selectTag(tags);
+    if (selectedTag == "BACK") {
+      running = false;
+    } else {
+      const untaggedTenant: Array<MeshTenant> | undefined = [];
+
+      for (const tenant of data) {
+        if (filterForTag(options, tenant, selectedTag) == false) {
+          untaggedTenant.push(tenant);
+        }
+      }
+      let runningInside = true;
+      while (runningInside) {
+        const selectedTenant = await selectTenant(
+          untaggedTenant,
+          undefined,
+          sortByCost,
+        );
+        if (selectedTenant == "BACK") {
+          runningInside = false;
+        } else {
+          await detailViewTenant(
+            options,
+            data,
+            selectedTenant,
+            selectedTag,
+            from == undefined || to == undefined,
+          );
+        }
+      }
+    }
+  }
+}
+
+function filterForTag(
+  _options: CmdGlobalOptions,
+  tenant: MeshTenant,
+  tagName: string,
+) {
+  let result = false;
+  for (const tag of tenant.tags) {
+    if (tag.tagName == tagName) {
+      result = true;
+    }
+  }
+  return result;
+}
+
+async function selectTag(
+  tags: Array<string>,
+  additionalOptions: Array<string> = [],
+) {
+  const options: Array<promptoptions> = [];
+  const help =
+    "\n\n\nHere you can select a tag, which should be missing on the tenants shown in the next step. This allows collie to filter all tenants, which allready have this tag assigned, from the selection in the next prompt. \n\n\n";
+
+  for (const tag of tags) {
+    options.push({ value: tag, name: tag });
+  }
+  for (const additionalOption of additionalOptions) {
+    options.push({ value: additionalOption, name: additionalOption });
+  }
+  options.push({ value: "HELP", name: "HELP" });
+  options.push({ value: "BACK", name: "BACK" });
+  options.push({ value: "QUIT", name: "QUIT" });
+
+  let selection = await Select.prompt({
+    message: "Select a tag",
+    options: options,
+  });
+  if (selection == "QUIT") {
+    Deno.exit();
+  } else if (selection == "BACK") {
+    console.clear();
+  } else if (selection == "HELP") {
+    console.clear();
+    console.log(help);
+    selection = await selectTag(tags, additionalOptions);
+  }
+  return selection;
+}
+
+async function selectTenant(
+  tenants: Array<MeshTenant>,
+  additionalOptions: Array<string> = [],
```

Comment:
t: you can actually write string[] = []; thats a bit shorter and means the same.